napari/layers/_base_layer/view/properties.py

```diff
@@ -26,32 +42,41 @@ def __init__(self, layer):
         cb.setProperty('mode', 'visibility')
         cb.stateChanged.connect(lambda state=cb: self.changeVisible(state))
         self.visibleCheckBox = cb
-        self.grid_layout.addWidget(cb, 0, 0)
+        self.top_layout.addWidget(cb)
+
+        tb = QLabel(self)
+        tb.setObjectName('thumbmnail')
+        tb.setToolTip('Layer thumbmnail')
+        self.thumbnail_label = tb
+        self._on_thumbnail_change(None)
+        self.top_layout.addWidget(tb)
 
         textbox = QLineEdit(self)
         textbox.setText(layer.name)
         textbox.home(False)
         textbox.setToolTip('Layer name')
-        textbox.setFixedWidth(122)
+        #textbox.setFixedWidth(122)
         textbox.setAcceptDrops(False)
         textbox.setEnabled(True)
         textbox.editingFinished.connect(self.changeText)
         self.nameTextBox = textbox
-        self.grid_layout.addWidget(textbox, 0, 1)
+        self.top_layout.addWidget(textbox)
 
-        self.grid_layout.addWidget(QLabel('opacity:'), 1, 0)
+        row = self.grid_layout.rowCount()
+        self.grid_layout.addWidget(QLabel('opacity:'), row, 0)
```

Comment:
I would suggest: 

```suggestion
        self.grid_layout.addWidget(QLabel('opacity:'), row=row, column=0)
```

but happy to drop it here. =)

Reply:
unfortunately I then get an error `row` is not a valid keyword argument. I will change it to
```python
row = self.grid_layout.rowCount()
column = 0
self.grid_layout.addWidget(QLabel('opacity:'), row, column)
```
and call it a day as I think it's then clear enough

Reply:
actually, I'm defining the following two variables
```python
self.name_column = 0
self.property_column = 1
```
and using them. can always fix in a followup PR